services/QuillLMS/engines/comprehension/app/models/comprehension/feedback.rb

```diff
@@ -20,5 +22,31 @@ def serializable_hash(options = nil)
         include: [:highlights]
       ))
     end
+
+    private def semantic_rule
+      rule.rule_type == Rule::TYPE_AUTOML
+    end
+
+    private def first_order
+      order == 0
+    end
+
+    private def second_order
+      order == 1
+    end
+
+    def log_update(user_id, prev_value)
+      if semantic_rule && first_order
+        rule&.prompts&.each do |prompt|
+          log_change(user_id, :update_feedback_1, prompt, nil, nil, prev_value, "#{rule.label.name} | #{rule.name}\n#{text}")
```

Comment:
This may be something that's actually intended, but won't this cause massive duplication if used for a universal rule?  It already causes a lot of duplication for just a Regex rule that applies to all three prompts, where things get logged three times, but if a universal rule gets this update, it'll get dozens of duplicates.